lib/colorpicker.go

```diff
@@ -0,0 +1,72 @@
+package lib
+
+import (
+	"math"
+
+	. "github.com/jedib0t/go-pretty/v6/text"
+)
+
+// FIXME: REFACTOR THIS. it was written by chatty jeeps since i didnt know how to implement this
```

Comment:
This is probably the best way to go about it, unmarshalling onto a struct is better than trying to parse manually, it also provides better type safety